app/components/pet_avatar_component.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Renders a User's avatar as image or user's initials
+class PetAvatarComponent < ApplicationComponent
+  option :pet, Types::Instance(Pet), optional: true
+  option :like, optional: true
+
+  private
+
+  def pet_avatar
```

Comment:
And I think this would better to be
pseudocode:
```
if pet_image exists, then return that image
otherwise, create the initials
```
I think this woul not render the initials if the pet did not have an image right now?